test/integration/reasoner/ResolutionTest.java

```diff
@@ -0,0 +1,451 @@
+                transaction.query().insert(Graql.parseQuery("insert $p1 isa person, has age 24; $t(twin1: $p1, twin2: $p2) isa twins; $p2 isa person;"));
+                transaction.commit();
+            }
+        }
+        long conjunctionTraversalAnswerCount = 3L;
+        long atomic1TraversalAnswerCount = 3L;
+        long atomic2TraversalAnswerCount = 3L;
+        long answerCount = conjunctionTraversalAnswerCount + (atomic2TraversalAnswerCount * atomic1TraversalAnswerCount);
+        Conjunction conjunctionPattern = parseConjunction("{ $t(twin1: $p1, twin2: $p2) isa twins; $p1 has age $a; }");
+        createRootAndAssertResponses(conjunctionPattern, answerCount);
+    }
+
+    @Test
+    public void filteringConcludable() throws InterruptedException {
+        try (RocksSession session = schemaSession()) {
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().define(Graql.parseQuery(
+                        "define person sub entity, owns age, plays twins:twin1, plays twins:twin2;" +
+                                "age sub attribute, value long;" +
+                                "twins sub relation, relates twin1, relates twin2;"));
+                transaction.commit();
+            }
+        }
+        try (RocksSession session = dataSession()) {
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().insert(Graql.parseQuery("insert $p1 isa person, has age 24;"));
+                transaction.query().insert(Graql.parseQuery("insert $p1 isa person, has age 24;"));
+                transaction.query().insert(Graql.parseQuery("insert $p1 isa person, has age 24;"));
+                transaction.commit();
+            }
+        }
+        long atomic1TraversalAnswerCount = 3L;
+        long atomic2TraversalAnswerCount = 0L;
+        long conjunctionTraversalAnswerCount = 0L;
+        long answerCount = conjunctionTraversalAnswerCount + (atomic1TraversalAnswerCount * atomic2TraversalAnswerCount);
+        Conjunction conjunctionPattern = parseConjunction("{ $t(twin1: $p1, twin2: $p2) isa twins; $p1 has age $a; }");
+        createRootAndAssertResponses(conjunctionPattern, answerCount);
+    }
+
+    @Ignore // TODO Un-ignore, ignored until rules are ready to use
+    @Test
+    public void simpleRule() throws InterruptedException {
+        // TODO This is what the initial test described, but since atomic2 is the conjunction the conjuntion traversal must return the same number of results as the atomic2 traversal.
+        String atomic1 = "$p1 isa person, has name \"Bob\";";
+        String atomic2 = "$p1 isa person, has age 42;";
+        String rulePattern = "rule bobs-are-42: when { $p1 isa person, has name \"Bob\"; } then { $p1 isa person, has age 42; };";
```

Comment:
errr this isn't a valid rule pattern?? is having `isa` on `p1` valid in graql??

Reply:
Good spot